app/Http/Controllers/FreeTrailController.php

```diff
@@ -0,0 +1,199 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Controllers\Order\BaseOrderController;
+use App\Model\Common\StatusSetting;
+use App\Model\Order\Invoice;
+use App\Model\Order\InvoiceItem;
+use App\Model\Order\Order;
+use App\Model\Payment\Plan;
+use App\Model\Product\Product;
+use App\Model\Product\Subscription;
+use App\User;
+use Auth;
+use Crypt;
+use DB;
+use Illuminate\Http\Request;
+use Lang;
+
+class FreeTrailController extends Controller
+{
+    public function __construct()
+    {
+        $this->middleware('auth');
+        $invoice = new Invoice();
+        $this->invoice = $invoice;
```

Comment:
Avoiding unnecessary variable declaration. You can write $this->invoice = new Invoice();